addon/webextension/background/selectorLoader.js

```diff
@@ -80,21 +85,43 @@ this.selectorLoader = (function() {
     });
   };
 
+  function readFile(path) {
+    return fetch(path, { mode: 'same-origin' }).then((response) => {
+      return response.blob();
+    }).then(blob => {
+      let reader = new FileReader();
+      return new Promise((resolve, reject) => {
+        reader.readAsText(blob);
+        reader.onload = (evt) => {
+          let contents = evt.target.result;
+          resolve(contents);
+        };
+        reader.onerror = (evt) => {
+          reject(evt.target.error);
+        };
+      });
+    });
+  }
+
   function executeModules(tabId, scripts) {
-    let lastPromise = Promise.resolve(null);
+    let filePromises = [];
     scripts.forEach((file) => {
-      lastPromise = lastPromise.then(() => {
-        return browser.tabs.executeScript(tabId, {
-          file,
-          runAt: "document_end"
-        }).catch((error) => {
-          log.error("error in script:", file, error);
-          error.scriptName = file;
-          throw error;
-        });
-      });
+      let fileURL = browser.extension.getURL(file);
+      let fileContents = readFile(fileURL);
+      filePromises.push(fileContents);
     });
-    return lastPromise.then(() => {
+    return Promise.all(filePromises).then(fileContents => {
+      let allScripts = fileContents.reduce((concatted, script) => { return concatted + ' ; ' + script });
```

Comment:
One bummer here is that we no longer get filenames or line positions. This is going to make debugging and Sentry much harder to work with.

I guess we could do a sourcemap? Though if we do, then using something like browserify (and doing this all as a make step) might be advantageous. I did a quick lookup of the sourcemap format, since maybe for pure concatenation it isn't that hard, but it looks pretty complicated. Though *maybe* it's not that complicated, since it looks like something like `{offset: {line: 10}, map: {file: "aScript.js"}}}` might work?